### PR TITLE
Boulder refineries can process golems

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

Golems (if lying down) can enter (and be processed by) boulder refineries. 

## Why It's Good For The Game

Rocks are rocks

## Changelog

:cl: Melbert
add: Boulder refineries and smelters can refine Golems. 
/:cl:

### CODE REVIEW

File: code/modules/mining/boulder_processing/_boulder_processing.dm [hunk 1]
```diff
@@ -177,13 +179,40 @@
...
+	if(!maim_golem(rockman))
+		return FALSE
+	playsound(src, usage_sound, 50, TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
+	COOLDOWN_START(src, accept_cooldown, 3 SECONDS)
+	return TRUE
+
+/// What effects actually happens to a golem when it is "processed"
+/obj/machinery/bouldertech/proc/maim_golem(mob/living/carbon/human/rockman)
+	Shake(duration = 1 SECONDS)
+	rockman.visible_message(span_warning("[rockman] is proccessed by [src]!"), span_userdanger("You get processed into bits by [src]!"))
```
Comment (mc-oofert) [inline code/modules/mining/boulder_processing/_boulder_processing.dm:201]:
```suggestion
	rockman.visible_message(span_warning("[rockman] is processed by [src]!"), span_userdanger("You get processed into bits by [src]!"))
```

File: code/modules/mining/boulder_processing/refinery.dm [hunk 2]
```diff
@@ -85,3 +85,9 @@
 /obj/machinery/bouldertech/refinery/smelter/on_set_is_operational(old_value)
 	set_light_on(TRUE)
 
+/obj/machinery/bouldertech/refinery/smelter/maim_golem(mob/living/carbon/human/rockman)
+	rockman.visible_message(span_warning("[rockman] is proccessed by [src]!"), span_userdanger("You get melted into rock by [src]!"))
```
Comment (MrMelbert) [inline code/modules/mining/boulder_processing/refinery.dm:89]:
```suggestion
	rockman.visible_message(span_warning("[rockman] is processed by [src]!"), span_userdanger("You get melted into rock by [src]!"))
```

**Comment by deathrobotpunch** [comment]
will this lead to funky material duping?

**Comment by MrMelbert** [comment]
No

**Pull request merged**